toonz/sources/toonz/filebrowserpopup.cpp

```diff
@@ -645,7 +645,51 @@ bool SaveSceneAsPopup::execute() {
           fp.getName())))  // Lol. Cmon! Really necessary?
     return false;
 
+#ifdef FEATURE_copyLevelsOnSaveSceneAs
+  // copy and replace all levels if project option "Separate assets into scene sub-folders" is checked
+  TProjectP project = TProjectManager::instance()->getCurrentProject();
+  if (project->getUseSubScenePath()) {
+    ToonzScene *scene = TApp::instance()->getCurrentScene()->getScene();
+    TLevelSet *levelSet = scene->getLevelSet();
+    for (int i = 0; i < levelSet->getLevelCount(); i++) {
+      TXshLevel *lvl = levelSet->getLevel(i);
+      auto pathQString = lvl->getPath().getQString();
+      QString oldSceneName = QString::fromStdString(scene->getScenePath().getName());
+      QString newSceneName = QString::fromStdString(fp.getName());
+      std::cout << "SaveSceneAsPopup: oldSceneName: " 
+                << oldSceneName.toStdString()
+                << " newSceneName : " << newSceneName.toStdString()
+                << std::endl;
+      if (oldSceneName != newSceneName) {
+        pathQString = pathQString.replace(oldSceneName, newSceneName);
+        std::cout << "SaveSceneAsPopup: saveLevel to "
+                  << pathQString.toStdString() << std::endl;
+        auto newPath = TFilePath(pathQString);
+        // 1 - Save level to a folder using new scene name
+        IoCmd::saveLevel(newPath, lvl->getSimpleLevel(), true);
+        // 2 - Update level's path property
+        lvl->getSimpleLevel()->setPath(newPath);
+      } else {
+        std::cout << "SaveSceneAsPopup: oldSceneName == newSceneName, keeping level paths intact "
+                  << std::endl;
+      }
+    }
+    // 3 - Notify xsheet scene and level of this change, of crash sooner or later ?
+    TApp::instance()->getCurrentXsheet()->notifyXsheetChanged();
+    TApp::instance()->getCurrentScene()->notifyCastChange();
+    TApp::instance()->getCurrentLevel()->notifyLevelChange();
+  }
+#endif
+#ifdef DEBUG_saveSceneAs
+  bool result = IoCmd::saveScene(fp, 0);
+  // automatic "save all" (otherwize, unsaved level changes would be lost)
```

Comment:
I don't think forcing a `saveAll()` after a `saveScene()` is necessary. There might be reasons the user wishes to save just the scene file as a different name and not save level changes.

After saveScene(), unsaved levels are still flagged as dirty, so the user will be forced to either save all or abandon the changes.

Reply:
Hello John,
thanks for the suggestion.
What about a popup on "save scene as", to give user the choice to save levels ?
It happened to many many times getting red drawings, because in other softwares save scene saves everything.
Actually levels are also saved anyway if (project->getUseSubScenePath()), but level->setPath should work following your logic ? (dirty state for future user saveAll).
Tell me if i am supposed to write those changes (well i am not sure how to edit code in the same PR).
Thank you !

Reply:
> What about a popup on "save scene as", to give user the choice to save levels ?

One of the reasons we moved `Save Scene` and `Save Scene As` into a sub-menu was to avoid confusion since we want people to use `Save All`.   I'd rather not introduce a pop-up for advance users who are more likely to be using this option instead of new users.

> Actually levels are also saved anyway if (project->getUseSubScenePath()), but level->setPath should work following your logic ? (dirty state for future user saveAll).

Yes, I saw you were saving it.  If setPath doesn't make the level dirty, you can always `setDirty(true)`

> Tell me if i am supposed to write those changes (well i am not sure how to edit code in the same PR).

Just update the branch you made your changes in and push the changes up and it will update the PR automatically.

Note: I noticed you used your `master` branch.  You should create a separate branch for changes and not touch your master.  This way, if you want to make other changes, they won't include stuff that either has not been officially merged yet since things can be merged in any order.